src/IdObfuscator/IdObfuscator.php

```diff
@@ -0,0 +1,69 @@
+<?php
+namespace Craftwork\IdObfuscator;
+
+final class IdObfuscator
+{
+    const CHARSET_DEFAULT = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_';
+
+    /**
+     * @var string
+     */
+    private $characterSet;
+
+    /**
+     * @var int
+     */
+    private $characterSetLength;
+
+    /**
+     * @param string $salt Salt that should be unique and kept secret for your application.
+     * @param string $characterSet An optional character set.
+     */
+    public function __construct(string $salt, string $characterSet = self::CHARSET_DEFAULT)
```

Comment:
That's a good point. It won't work properly if the character set has duplicates. And it also need to be at least 2 characters long, it would be useless to use base 1 anyway.

The salt can be any string though..